### PR TITLE
Issue #437 - Add Ability to Hide Cancelled Arrangements

### PR DESCRIPTION
**Pull request by donschaefer**

### CODE REVIEW

File: src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx [hunk 1]
```diff
@@ -80,6 +82,68 @@ export function PartneringFamilyScreen() {
...
+      }
+      return o;
+    }));
+  }
+
+  const meetsArrangementFilterCriteria = (arrangement: Arrangement): boolean => {
+    // TODO: Reduce the number of times this method fires on the initial page load
+    let result = false;
+    const selectedOptions = arrangementFilterOptions.filter(o => o.selected).map(o => o.text);    
+    if (selectedOptions.length) {
```
Comment (LarsKemmann) [inline src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx:125]: I think this `if` is unnecessary -- if `.length` is zero, `.forEach(...)` will just exit without doing anything.

File: src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx [hunk 2]
```diff
@@ -80,6 +82,68 @@ export function PartneringFamilyScreen() {
...
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const handleFilterClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleFilterClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'Filter' : undefined;
```
Comment (LarsKemmann) [inline src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx:95]: I would recommend renaming `open` and `id` to be a bit more clear in the context of this (admittedly) way-too-large function, but I'm half-guessing that you're thinking about extracting the filter to its own component? That would be better (also because then we're starting to slim down the `PartneringFamilyScreen`).

File: src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx [hunk 3]
```diff
@@ -228,16 +292,48 @@ export function PartneringFamilyScreen() {
...
+                        vertical: 'bottom',
+                        horizontal: 'left',
+                      }}
+                    >
+                      <FormControl sx={{ m: 3 }} component="fieldset" variant="standard">
+                        <FormLabel component="legend">Filter Arrangements</FormLabel>
+                        <FormGroup>
+                          {
+                          arrangementFilterOptions.map((o) => {
+                            return (
```
Comment (LarsKemmann) [inline src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx:321]: You actually don't need the `{ return ( /*...*/ ); }` -- you can just do something like:
```
<FormGroup>
  {arrangementFilterOptions.map(o => (
    <FormControlLabel /*...*/ />
  ))}
</FormGroup>
```
(and I'm even not sure if the inner parentheses are needed)

**Review by LarsKemmann — CHANGES_REQUESTED** [review]
Nice work @donschaefer! This runs great locally. I just had a couple notes on code style/organization 1) for consistency, because I still haven't implemented automated code formatting via Prettier (#207), and 2) because this component, like several of the others I had created, is already embarrassingly large. 😳

File: src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx (hunk 1 above)
Comment (donschaefer) [reply src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx]: Fixed

File: src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx (hunk 2 above)
Comment (donschaefer) [reply src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx]: This is now fixed & also moved to it's own component over @ FilterMenu.tsx

File: src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx (hunk 3 above)
Comment (donschaefer) [reply src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx]: Thanks for the suggestion - Fixed

File: src/caretogether-pwa/src/Families/FamilyScreen.tsx [hunk 4]
```diff
@@ -106,6 +108,34 @@ export function FamilyScreen() {
...
+  enum ArrangementFilterOptionLabel {
+    Active = "Active",
+    Cancelled = "Cancelled"
+  }
+  const { arrangementFilterOptions, handleFilterArrangements } =
+    useFilterMenu(Object.keys(ArrangementFilterOptionLabel), [ArrangementFilterOptionLabel.Active]);
+  const meetsArrangementFilterCriteria = (arrangement: Arrangement): boolean => {
+    let result = false;
+    const selectedOptions = arrangementFilterOptions.filter(o => o.selected).map(o => o.text);
+    selectedOptions.forEach((option) => {
```
Comment (LarsKemmann) [inline src/caretogether-pwa/src/Families/FamilyScreen.tsx:121]: This code could be simplified to just `const selectedOptions = ....filter(...).map(...).any(o => ...)` (so no `forEach` is needed).

**Review by LarsKemmann — CHANGES_REQUESTED** [review]

File: src/caretogether-pwa/src/Families/FamilyScreen.tsx (hunk 4 above)
Comment (LarsKemmann) [reply src/caretogether-pwa/src/Families/FamilyScreen.tsx:121]: The reason to do this is that it would also get rid of the `let result = false` (avoiding mutable state) and the need for things like `break` statements (which can be hard to follow).

Comment (donschaefer) [reply src/caretogether-pwa/src/Families/FamilyScreen.tsx]: Great suggestion - implemented! Also tweaked the UI to collapse when there were no matching results since I couldn't think of a valid use case for persisting empty placeholder space. 

To be clear, it already starts out collapsed if there aren't any results, but... prior to my latest update, if you add arrangements to the mix & then adjust the filters to filter them back out, Masonry was retaining whatever the last height was before the results went from a length of > 0 to a length of 0. This update makes it so that when there are 0 results, it ALWAYS collapses.

That said, I'm not _thrilled_ with the inline style tag I used to accomplish that UX improvement, so... if you have suggestions on a better approach there, I'm happy to take a pointer.

**Review by LarsKemmann — APPROVED** [review]
Nice work! I'm good with the inline style tag; at some point soon this screen needs a full design overhaul and the masonry component is one thing I'm hoping to get rid of by swapping it for a more table-based layout with side-sheet editing. (Which is partly why I'm thrilled at how reusable this filtering logic is. :))
